pkgs/by-name/cr/creek/package.nix

```
@@ -12,43 +12,44 @@
   wayland-protocols,
 }:
 let
-  zig = zig_0_13;
+  zig = zig_0_14;
 in
 stdenv.mkDerivation (finalAttrs: {
   pname = "creek";
-  version = "0.4.2";
+  version = "0.4.3";
 
   src = fetchFromGitHub {
     owner = "nmeum";
     repo = "creek";
     tag = "v${finalAttrs.version}";
-    hash = "sha256-3Q690DEMgPqURTHKzJwH5iVyTLvgYqNpxuwAEV+/Lyw=";
+    hash = "sha256-5TANQt/VWafm6Lj4dYViiK0IMy/chGr/Gzq0S66HZqI=";
   };
 
```

```suggestion
  depsBuildBuild = [
    pkg-config
  ];
```
This is required for `strictDeps = true` support (#178468).

LGTM otherwise :)

---

Thanks for the feedback—it caught me off guard. You never know where the next rabbit hole lies.

---

Oh, you do need both, actually :wink: 

1. `depsBuildBuild`'s pkg-config is required to find wayland-scanner
2. `nativeBuildInputs`'s pkg-config is required to find wayland-protocols

As for why, perhaps it helps to know that "nativeBuildInputs" would be called `depsBuildHost` in the generalized naming scheme.
`depsXY` specify packages that run on X and "target" Y in some way.
So `depsBuildBuild`'s pkg-config will know about the build platform (to get wayland-scanner, here), and `depsBuildHost`'s pkg-config will know about the host platform's libraries (to find wayland-protocols, here).

But this whole distinction is thrown out of the window (by default) for native builds. The default `strictDeps = false;` essentially merges nativeBuildInputs and buildInputs into one big pile. Leading to lots of trivially-fixed failures when trying to cross-compile.

---

This explanation could have spared me some days of research and learning about "offsets". But here I still missed the issue. Is there a foolproof test to help detect a dependency missing or in a wrong place?

---

Yes, add `strictDeps = true;` to the package / try to cross-compile it. The latter will take considerably more time as most cross-compiled packages are not cached. The former will only take as long as a normal build.

---

Thank you, I will note it for the future.